src/test/client/apiclient.spec.unit.ts

```diff
@@ -1,70 +1,74 @@
-const mockRequest: jest.Mock = jest.fn( () => { return dummyAxiosResponse });
-jest.mock("axios", () => {
-  return {
-    default: {
-      request: mockRequest
-    }
-  };
-});
+import {getCompanyProfile, ExtensionsCompanyProfile} from "../../client/apiclient";
+import * as mockUtils from "../mock.utils";
+import Resource from "ch-sdk-node/dist/services/resource";
+import {CompanyProfile} from "ch-sdk-node/dist/services/company-profile";
+import CompanyProfileService from "ch-sdk-node/dist/services/company-profile/service";
 
-beforeEach( () => {
-  mockRequest.mockClear();
-});
+//////////////////
+//Set up mocks
 
-// Need to import after mocks set or the real axios module will be imported before mocks
-import {AxiosResponse} from "axios";
-import {CompanyProfileResource, getCompanyProfile, CompanyProfile} from "../../client/apiclient"
-import * as mockUtils from "../mock.utils";
+jest.mock("ch-sdk-node/dist/services/company-profile/service");
 
-const dummyAxiosResponse: AxiosResponse<CompanyProfileResource> =  {
-  data: {
-    accounts: {
-      next_due: "2020-05-31",
-      overdue: false,
-      next_accounts: {
-        period_end_on: "2019-10-10",
-        period_start_on: "2019-01-01",
-      },
-    },
-    company_number: "00006400",
-    company_name: "Girl's school trust",
-    company_status: "active",
-    date_of_creation: "1872-06-26",
-    has_been_liquidated: false,
-    has_charges: false,
-    has_insolvency_history: false,
-    jurisdiction: "england",
-    type: "limited",
-    registered_office_address: {
-      address_line_1: "line1",
-      address_line_2: "line2",
-      postal_code: "post code"
-    }
-  },
-  status: 200,
-  statusText: "OK",
-  headers: "header",
-  config: {}
-};
+// end of set up mocks
+///////////////////////
 
-describe("apiclient unit tests", () => {
+describe("apiclient company profile unit tests", () => {
+  const mockGetCompanyProfile = (CompanyProfileService.prototype.getCompanyProfile as jest.Mock);
+  beforeEach(() => {
+    mockGetCompanyProfile.mockReset();
+  });
 
   it("converts company number to uppercase", async () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
     const company = await getCompanyProfile("sc100079", mockUtils.ACCESS_TOKEN);
     expect(company.incorporationDate).toEqual("26 June 1872");
-    const args = mockRequest.mock.calls[0][0];
-    expect(args.url).toContain("SC100079");
+    const args = mockGetCompanyProfile.mock.calls[0][0];
```

Comment:
can we change this to use `expect().toBeCalledWith()`?

```typescript
    expect(mockGetCompanyProfile).toBeCalledWith("SC100079");
```

it describes the assertion better than accessing the param via array indexes.